OPTIMIZATION: Storage, Capacity, and Prerequisites (#1118)
minio/docs: source/operations/checklists/hardware.rst

The following checklist follows MinIO's `Recommended Configuration <https://min.io/product/reference-hardware?ref-docs>`__ for production deployments.
The provided guidance is intended as a baseline and cannot replace |subnet| Performance Diagnostics, Architecture Reviews, and direct-to-engineering support.

> MinIO strongly recommends selecting substantially similar hardware configurations for all nodes in a given :term:`server pool`. 

Reviewer: Maybe an allusion to the fact this is desired for any distributed system

> Ensure a consistent selection hardware (CPU, memory, motherboard, storage adapters) and software (operating system, kernel settings, system services) across pool nodes. 

Reviewer 2: probably
```suggestion
Ensure a consistent selection of hardware (CPU, memory, motherboard, storage adapters) and software (operating system, kernel settings, system services) across pool nodes. 
```

> Deployment may exhibit unpredictable performance if nodes have varying hardware or software configurations. 

Reviewer 2: ```suggestion
Deployments may exhibit unpredictable performance if nodes have varying hardware or software configurations. 
```

.. cond:: not k8s

   Recommended Storage Mediums
   +++++++++++++++++++++++++++

   MinIO recommends using flash-based storage (NVMe or SSD) for all workload types and scales.
   Workloads that require high performance should prefer NVMe over SSD.

   MinIO deployments using HDD-based storage are best suited as cold-tier targets for :ref:`Object Transition ("Tiering") <minio-lifecycle-management-tiering>` of aged data.
   HDD storage typically does not provide the necessary performance to meet the expectations of modern workloads, and any cost efficiencies at scale are offset by the performance constraints of the medium. 

   Use Direct-Attached "Local" Storage (DAS)
   +++++++++++++++++++++++++++++++++++++++++

>    DAS-storage, such as locally-attached JBOD (Just a Bunch of Disk) arrays, provide significant performance and consistency advantages over networked (NAS, SAN, NFS) storage.

Reviewer 2: Is `DAS-storage` common usage? To me, it reads as `Direct-Attached Storage-Storage`

   .. dropdown:: Network File System Volumes Break Consistency Guarantees
      :class-title: note

      MinIO's strict **read-after-write** and **list-after-write** consistency
      model requires local drive filesystems.

      MinIO cannot provide consistency guarantees if the underlying storage
      volumes are NFS or a similar network-attached storage volume. 

>       For deployments that *require* using network-attached storage, use

Reviewer: I don't like the idea of saying to use NFS at all, and v4 only mitigates cache loss in most scenarios, not all. Would just say dont or not recommended and leave it at that .

   Using any other type of backing storage (SAN/NAS, ext4, RAID, LVM) typically results in a reduction in performance, reliability, predictability, and consistency.

   When formatting XFS drives, apply a unique label per drive.
   For example, the following command formats four drives as XFS and applies a corresponding drive label.

   .. code-block:: shell

      mkfs.xfs /dev/sdb -L MINIODRIVE1
      mkfs.xfs /dev/sdc -L MINIODRIVE2
      mkfs.xfs /dev/sdd -L MINIODRIVE3
      mkfs.xfs /dev/sde -L MINIODRIVE4

   Mount Drives using ``/etc/fstab``
   +++++++++++++++++++++++++++++++++

   MinIO **requires** that drives maintain their ordering at the mounted position across restarts.
   MinIO **does not** support arbitrary migration of a drive with existing MinIO data to a new mount position, whether intentional or as the result of OS-level behavior.

   You **must** use ``/etc/fstab`` or a similar mount control system to mount drives at a consistent path.
   For example:

   .. code-block:: shell
      :class: copyable

      $ nano /etc/fstab

      # <file system>        <mount point>    <type>  <options>         <dump>  <pass>

>       LABEL=MINIODRIVE1      /mnt/drive-1     xfs     defaults,noatime  0       2

Reviewer: technically we don't need noatime anymore but it doesn't hurt to have it

      LABEL=MINIODRIVE2      /mnt/drive-2     xfs     defaults,noatime  0       2
      LABEL=MINIODRIVE3      /mnt/drive-3     xfs     defaults,noatime  0       2
      LABEL=MINIODRIVE4      /mnt/drive-4     xfs     defaults,noatime  0       2

>    You can then use ``mount -a`` or similar automounting configurations to ensure all MinIO drives mount at node startup.

Reviewer: nit: mount -a is not an automount

   MinIO **strongly recommends** using label-based mounting rules over UUID-based rules.
   Label-based rules allow swapping an unhealthy or non-working drive with a replacement that has matching format and label.
   UUID-based rules require editing the ``/etc/fstab`` file to replace mappings with the new drive UUID.

   .. note:: 

      Cloud environment instances which depend on mounted external storage may encounter boot failure if one or more of the remote file mounts return errors or failure.

>       For example, an AWS ECS instances with mounted persistent EBS volumes may fail to boot with the standard ``/etc/fstab`` configuration if one or more EBS volumes fail to mount.

Reviewer 2: If you like: tiny tweak to remove one of the many variations on `fail` here
```suggestion
      For example, an AWS ECS instances with mounted persistent EBS volumes may not boot with the standard ``/etc/fstab`` configuration if one or more EBS volumes fail to mount.
```

Reviewer 3: Note: we'd need to drop `an` or make `instance` singular.

      You can set the ``nofail`` option to silence error reporting at boot and allow the instance to boot with one or more mount issues.
      

>       You should not use this option on systems which have locally attached disks, as silencing drive errors prevents both MinIO and the OS from responding to those errors in a normal fashion.

Reviewer 2: perhaps
```suggestion
      You should not use this option on systems with locally attached disks, as silencing drive errors prevents both MinIO and the OS from responding to those errors in a normal fashion.
```

   Use Direct-Attached "Local" Storage (DAS)
   +++++++++++++++++++++++++++++++++++++++++

>    :abbr:`DAS (Direct-Attached Storage)`, such as locally-attached JBOD (Just a Bunch of Disk) arrays, provide significant performance and consistency advantages over networked (NAS, SAN, NFS) storage.

Reviewer 2: It's `Disks` right? Dunno why I can't suggest.

Reviewer 2: ok, all commits view lets me suggest
```suggestion
   :abbr:`DAS (Direct-Attached Storage)`, such as locally-attached JBOD (Just a Bunch of Disks) arrays, provide significant performance and consistency advantages over networked (NAS, SAN, NFS) storage.
```
